### ISSUE TITLE
File links failing? 

### ISSUE THREAD

**Issue by AlexWatterson** [brendanheywood/moodle-local_cleanurls#47]
Hi,
Thanks for all the work on CleanURLs, quick query which hopefully has simple answer:
I just installed clean_urls on Moodle 3.2 test platform. All OK i.e. 
"Rewrite router is working (inbound links)
Rewrite function is configured properly (outbound links)"
All Moodle "native" content in courses i.e .forum, announcements etc... work BUT links to files (Word/PPT etc..) dont! E.g. previously say http://myServer/mod/resource/view.php?id=4668 but now http://myserver/course/PX904+1617/resource/4668-lecture-0 does not work.
I checked and makes no difference if course created before or after clearnurl installed.

Checked error log and (changed IP and Server to tags)

```
[Tue Jan 31 13:55:40.921695 2017] [proxy_fcgi:error] [pid 22833:tid 140150245820160] [client myIP:1523] AH01071: Got error '
PHP message: Router: $_GET: 'course/PX904+1617/resource/4668-lecture-0'\n
PHP message: Incoming url: http://myServer/course/PX904+1617/resource/4668-lecture-0\n
PHP message: Incoming path: /course/PX904+1617/resource/4668-lecture-0\n
PHP message: Rewritten to: /mod/resource/view.php\n
PHP message: Rewritten to: /mod/resource/view.php\n
PHP message: Router: including file: /var/www/html/mod/resource/view.php\n
PHP message: PHP Notice:  Undefined property: stdClass::$dirroot in /var/www/html/mod/resource/view.php on line 27\n
PHP message: PHP Warning:  require_once(/mod/resource/lib.php): failed to open stream: No such file or directory in /var/www/html/mod/resource/view.php on line 27\n
', referer: http://myServer/course/PX904+1617
```

If I switch cleanURLS off all links work (e.g. http://myServer/mod/resource/view.php?id=4668 downloads the file) so definitely not some kind of file/permissions issue.

Any ideas?

Thanks,
Alex

**Comment by brendanheywood** [comment]
Yes this is likely the very few first lines of your config.php file, please doble check it is exactly setup like this:

https://github.com/brendanheywood/moodle-local_cleanurls#step-4-turn-it-on-and-configure

Note this is different to what is there by default in config-dist.php

**Comment by AlexWatterson** [comment]
Outstanding Brendan that did indeed resolve it :-) Many thanks for the prompt reply. 

I had added those changes however must have been corrupted possibly by "unset ($CFG)" in legacy config.php, rejigging order seems to have resolved it (current code below works fine). 

On a side note I've found any course names which contain brackets or slashes do not work i.e a course called say “LN301 (16/17)” fails i.e it turns it into: http://myserver/course/LN301+%2816%2F17%29 which throws a 500 error. We can work around this by removing all brackets or slashes but just wondered if this was likely to be resolved? I note earlier issue where even more obscure characters such as "Ã" were causing issues but wonder if brackets/slashes maybe quite widely used for course dates etc..

Thanks again for your help,

Alex

```
<?php  // Moodle configuration file
global $CFG;
// Below 3 lines for cleanurls
if (!isset($CFG)) {
    $CFG = new stdClass();
}
//unset($CFG);
//global $CFG;
//$CFG = new stdClass();
//$CFG->urlrewriteclass = '\local_cleanurls\url_rewriter';
$CFG->urlrewriteclass = '\\local_cleanurls\\url_rewriter';
```
